### PR TITLE
Prevent crash when calling lerp for IconThemeData with null arguments

### PR DESCRIPTION
**Pull request by rami-a**
The lerp method for IconThemeData was not doing anything to prevent exceptions when null arguments are used. This change fixes that and adds a test for it.

### CODE REVIEW

File: packages/flutter/test/widgets/icon_theme_data_test.dart [hunk 1]
```diff
@@ -18,4 +18,13 @@ void main() {
     expect(lerped.opacity, equals(0.625));
     expect(lerped.size, equals(18.0));
   });
+
+  test('IconThemeData lerp with null', () {
```
Comment (HansMuller) [inline packages/flutter/test/widgets/icon_theme_data_test.dart]: This is good.  Please add a test for IconThemeData.lerp(null, data, 0.25); and IconThemeData.lerp(null, null, 0.25).

**Review by HansMuller — CHANGES_REQUESTED** [review]
Nice fix!

File: packages/flutter/test/widgets/icon_theme_data_test.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/test/widgets/icon_theme_data_test.dart]: Good idea, done.

**Review by HansMuller — CHANGES_REQUESTED** [review]
The original test wasn't the perfect template. Sorry about that. Just a few more small tweaks..

File: packages/flutter/test/widgets/icon_theme_data_test.dart [hunk 2]
```diff
@@ -19,12 +19,28 @@ void main() {
     expect(lerped.size, equals(18.0));
   });
 
-  test('IconThemeData lerp with null', () {
+  test('IconThemeData lerp with first null', () {
+    const IconThemeData data = IconThemeData(color: Color(0xAAAAAAAA), opacity: 0.5, size: 16.0);
+
+    final IconThemeData lerped = IconThemeData.lerp(null, data, 0.25);
+    expect(lerped.color, equals(Color.lerp(null, const Color(0xAAAAAAAA), 0.25)));
+    expect(lerped.opacity, equals(0.125));
```
[the remark this answers is not in the document]
Comment (rami-a) [reply packages/flutter/test/widgets/icon_theme_data_test.dart]: Good to know, done.

File: packages/flutter/test/widgets/icon_theme_data_test.dart [hunk 3]
```diff
@@ -19,12 +19,28 @@ void main() {
     expect(lerped.size, equals(18.0));
   });
 
-  test('IconThemeData lerp with null', () {
+  test('IconThemeData lerp with first null', () {
+    const IconThemeData data = IconThemeData(color: Color(0xAAAAAAAA), opacity: 0.5, size: 16.0);
+
+    final IconThemeData lerped = IconThemeData.lerp(null, data, 0.25);
+    expect(lerped.color, equals(Color.lerp(null, const Color(0xAAAAAAAA), 0.25)));
```
[the remark this answers is not in the document]
Comment (rami-a) [reply packages/flutter/test/widgets/icon_theme_data_test.dart]: Done

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/icon_theme_data_test.dart [hunk 4]
```diff
@@ -14,8 +14,33 @@ void main() {
...
-    expect(lerped.color, equals(Color.lerp(const Color(0xAAAAAAAA), const Color(0xFF000000), 0.25)));
-    expect(lerped.opacity, equals(0.625));
-    expect(lerped.size, equals(18.0));
+    expect(lerped.color, const Color(0xBF7F7F7F));
+    expect(lerped.opacity, 0.625);
+    expect(lerped.size, 18.0);
+  });
+
+  test('IconThemeData lerp with first null', () {
+    const IconThemeData data = IconThemeData(color: Color(0xAAAAAAAA), opacity: 0.5, size: 16.0);
```
[the remark this answers is not in the document]
Comment (rami-a) [reply packages/flutter/test/widgets/icon_theme_data_test.dart]: Done

**Pull request merged**